pkg/statistics/builder.go

```diff
@@ -360,7 +370,7 @@ func BuildHistAndTopN(
 
 	// Handle the counting for the last value. Basically equal to the case 2 above.
 	// now topn is empty: append the "current" count directly
-	if numTopN != 0 {
+	if numTopN != 0 && (!allowPruning || (allowPruning && (sampleFactor <= 1 || curCnt > 1))) {
```

Comment:
Maybe we should add some comments to explain here we still try to ignore topn with count 1.